client/my-sites/plans-features-main/hooks/use-eligibility-for-term-savings-price-display.ts

```diff
@@ -75,11 +70,52 @@ const useEligibilityForTermSavingsPriceDisplay = ( {
 		false
 	);
 
-	if ( isAnyGridPlanDiscounted ) {
-		return false;
-	}
+	return isAnyGridPlanDiscounted;
+};
+
+const useEligibilityForTermSavingsPriceDisplay = ( {
+	hiddenPlans,
+	intent,
+	isSubdomainNotGenerated,
+	selectedPlan,
+	term,
+	displayedIntervals,
+	coupon,
+	siteId,
+	isInSignup,
+}: {
+	hiddenPlans?: HiddenPlans;
+	intent?: PlansIntent;
+	isSubdomainNotGenerated?: boolean;
+	selectedPlan?: PlanSlug;
+	term: ( typeof TERMS_LIST )[ number ];
+	displayedIntervals: UrlFriendlyTermType[];
+	coupon?: string;
+	siteId?: number | null;
+	isInSignup?: boolean;
+} ) => {
+	const isAnyGridPlanDiscounted = useIsAnyGridPlanDiscounted( {
+		hiddenPlans,
+		intent,
+		isSubdomainNotGenerated,
+		selectedPlan,
+		term,
+		displayedIntervals,
+		coupon,
+		siteId,
+	} );
+
+	const [ isLoading, experimentAssignment ] = useExperiment(
+		'calypso_plans_page_emphasize_longer_term_savings_v2',
+		{
+			isEligible: isInSignup && ! isAnyGridPlanDiscounted,
+		}
+	);
 
-	return longerPlanTermDefaultExperiment.isEligibleForTermSavings && isInSignup;
```

Comment:
In the current implementation, if `isAnyGridPlanDiscounted` is `true` or `isInSignup` is `false`, the user will still be assigned to the experiment, but will not see either experiment variation, skewing the results towards control. I have moved these checks to the `isEligible` param of `useExperiment` so that users who are not eligible will not be assigned.

Reply:
Ah makes sense. Thanks for catching this @aneeshd16 👍